### DISCUSSION TITLE
How to expose Prometheus metrics manually?

### DISCUSSION THREAD

**Discussion by PavelPenkov** [kamon-io/Kamon#1074] [Q&A] [answered]
Say I don't want to use NanoHTTP server and want to serve metrics using my "main" HTTP server. How could I do that?

**Comment by ivantopo** [comment, marked as answer]
In general, you would need to follow the steps outlined in https://github.com/kamon-io/Kamon/pull/1079 's description, but this question has come up so often that I went ahead for a simpler solution that will go out with Kamon 2.3.2:

Call `PrometheusReporter.latestScrapeData()` and you'll get back an `Option[String]`. If the Prometheus reporter is running then the option will be defined and contain the latest scrape data, otherwise it wont. On top of that, you could disable the built-in HTTP server using this setting:

```hcl
kamon.prometheus {

  # Enable or disable publishing the Prometheus scraping enpoint using a embedded server.
  start-embedded-http-server = no
}
```

Stay alert for the 2.3.2 release!
